packages/test/test-version-utils/src/describeCompat.ts

```diff
@@ -221,37 +227,36 @@ export type DescribeCompat = DescribeCompatSuite &
 function createCompatDescribe(): DescribeCompat {
 	const createCompatSuiteWithDefault = (
 		tests: (this: Mocha.Suite, provider: () => ITestObjectProvider, apis: CompatApis) => void,
-		compatVersionKind: CompatVersionKind,
+		compatVersion: string,
 	) => {
-		switch (compatVersionKind) {
+		switch (compatVersion) {
 			case "FullCompat":
 				return createCompatSuite(tests, undefined);
 			case "NoCompat":
```

Comment:
I think "NoCompat" should go away with this change. It should either be FullCompat or specify a min version.

Reply:
There are several issues with removing NoCompat:
1. All current NoCompat test would need to be changed to specify a min version. It will be very hard to know which version each suite was intended to be tested against in the first place.
2. Even if we do the job and specify a min version to all current NoCompat tests, curretn infrastructure won't support such a change. Scott and I have tried to change current NoCompat tests that already work FullCompat and pipeline timeout is almost guaranteed. 
3. Current rc version and 2.0.0-internal.8.0.0 are not getting correct N-i versions. (don't know if it is because current special rc version).
4. I would wait for the next release to tackle some of the issues described before and only once they are fixed, deprecate NoCompat.

Reply:
> Current rc version and 2.0.0-internal.8.0.0 are not getting correct N-i versions. (don't know if it is because current special rc version).

FYI, this should be fixed after https://github.com/microsoft/FluidFramework/pull/19092 / https://github.com/microsoft/FluidFramework/pull/19127 🙂

Reply:
Even if some of the issues are already fixed, I think we still need to keep NoCompat and slowly remove those to either a min version or full compat.

Reply:
It's okay if that doesn't happen in this PR. You can create a follow up task to tackle this.

> All current NoCompat test would need to be changed to specify a min version. It will be very hard to know which version each suite was intended to be tested against in the first place.

I wouldn't bother finding out when the test was added but use a recent version - maybe internal-7.0 or something. Eventually, these tests will start testing fullCompat.

Reply:
One desirable outcome of this PR (and any immediate follow-up) would be that new tests are not added with "NoCompat". With that in mind, do you think we should deprecate "NoCompat" option so that anyone adding it immediately knows that there is an alternative? Although for that you will have to keep `CompatVersionKind` and maybe add the `minVersion` as a separate parameter. Maybe that's okay?